__tests__/lib/edge-runtime/angelaUtils.test.ts

```diff
@@ -0,0 +1,137 @@
+import { updatePastAssistantMessage } from "../../../lib/edge-runtime/angelaUtils";
```

Comment:
Like this :D 

can you move streamResponseToUser and replacePlaceholdersDuringStreaming to angelaUtils?